src/CurrencyChanger

```diff
@@ -0,0 +1,96 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.text.DecimalFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CurrencyBalanceApp implements User {
```

Comment:
User is not an interface.  It cannot be implemented.  
Did you perhaps mean to extend it?
If so, you are extending the base class rather than getting the actual user once it gets set up.  The user's information will not be accessible using this method.